### ISSUE TITLE
convert blocks to js on server instead of browser

### ISSUE THREAD

**Issue by iasoon** [ZeusWPI/MOZAIC#89]
Blocked by https://github.com/mo4islona/node-blockly/issues/15 .
This way we don't have to send random js to the server.

**Comment by redfast00** [comment]
Related: I wrote a [small Python script](https://gist.github.com/redfast00/e61c3bf4687bb9ef3f4740fb0acc6ce9) for mitmproxy to intercept the generated JS and upload your own JS. This allows you to write in JS instead of in blockly.

**Comment by wschella** [comment]
@redfast00, while writing in more practical languages is certainly the way to go, and the way the architecture is designed. Like @iasoon said, executing arbitrary JS might not be very well from a security point of view, and it blocks us from promoting this link publicly. So we should update our server by solving this issue and rendering your script useless.

**Comment by redfast00** [comment]
Yes, by all means, please don't have the server run arbitrary JS. My script was just a quick PoC so I could program in JS for the time it worked.

**Comment by wschella** [comment]
The [blocking issue](https://github.com/mo4islona/node-blockly/issues/15) was recently closed, which means we could actually do this now. Not that it very relevant anymore now.
